### PR TITLE
feat: open goal modal with spacebar key

### LINKED ISSUE #2053: Open modal for goal with 'spacebar' key.
**Is your feature request related to a problem? Please describe.**
It's very annoying when you're used to using 'spacebar' to click on things - and it doesn't work.

**Describe the solution you'd like**
When navigating through goals using the keyboard, pressing 'spacebar' key will go into the highlighted goal, just like tapping/clicking.

**Describe alternatives you've considered**
Using 'enter' key - but this is reserved for quick-adding goals.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by harshalranjhani**
the goal modal now opens with spacebar key, a similar action which could be performed by clicking the goal as well

fixes #2053 

Screenshot:
image

### CODE REVIEW

**Comment by tijlleenders** [comment]
@harshalranjhani The original intent of the issue was to navigate 'into' the goal - instead of opening the modal...   
... but actually the way you naturally interpreted it probable makes more sense. Keep the left/right/up/down keys for navigation, and spacebar for 'action'.  


I'm good with it - except that the background navigates to a higher lever when one presses spacebar at a lower level:  
[Screencast from 2024-10-13 14-09-20.webm](https://github.com/user-attachments/assets/0a95e5f2-546a-478a-a3d5-3328760843ec)

**Comment by harshalranjhani** [comment]
> background navigates to a higher lever when one presses spacebar at a lower level

hii, i've fixed this with the new commit. let me know if i fixed the issue correctly.

**Review by tijlleenders — APPROVED** [review]
Awesome, thanks!

**Comment by harshalranjhani** [comment]
hey @tijlleenders , i was playing around with it when i realised that if the goal was a child goal, the state wasn't preserved and in the background, the parent goal title was disappearing, so i fixed that as well with this new commit and now everything should work perfectly fine!

**Review by tijlleenders — DISMISSED** [review]
Good catch!

File: src/components/GoalsComponents/GoalSublist/components/GoalHistory.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ const breadcrumbStyle: React.CSSProperties = {
 };
 
 const BreadcrumbItem = ({ goal }: { goal: ISubGoalHistory }) => {
+  console.log(goal.goalTitle);
```
Comment (vinaybadgujar102) [inline src/components/GoalsComponents/GoalSublist/components/GoalHistory.tsx:27]: remove this log

File: src/hooks/useGoalSelection.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export const useGoalSelection = (goals: GoalItem[]): GoalItem | undefined => {
   const downPress = useKeyPress("ArrowDown");
   const rightPress = useKeyPress("ArrowRight");
   const leftPress = useKeyPress("ArrowLeft");
+  const spacePress = useKeyPress(" ");
```
Comment (vinaybadgujar102) [inline src/hooks/useGoalSelection.ts:17]: can we use "Space" key code to detect the space key press?

**Review by vinaybadgujar102 — CHANGES_REQUESTED** [review]

File: src/hooks/useGoalSelection.ts (hunk 2 above)
Comment (harshalranjhani) [reply src/hooks/useGoalSelection.ts]: @vinaybadgujar102 fixed it.

File: src/components/GoalsComponents/GoalSublist/components/GoalHistory.tsx (hunk 1 above)
Comment (harshalranjhani) [reply src/components/GoalsComponents/GoalSublist/components/GoalHistory.tsx]: removed this as well.

**Review by vinaybadgujar102 — APPROVED** [review]

**Comment by vinaybadgujar102** [comment]
Thanks for your contribution and effort 🙌

**Pull request merged**